src/fixup.js

```diff
@@ -141,6 +141,29 @@
     }
   }
 
+  /* fragment identifiers backward compatibility mapping */
+  if (document.location.hash !== "") {
```

Comment:
Maybe:

```JS
if (
  location.hash &&
  !document.getElementById(location.hash) && // not broken link
  document.getElementById("fragment_id_mapping") // we can remap
) {
  var idMapElement = document.getElementById("fragment_id_mapping");
  try {
    var data = JSON.parse(idMapElement.textContent);
    var hash = location.hash.substr(1);
    var canonicalId = data.hasOwnProperty(hash) ? data[hash] : undefined;
    if (canonicalId) location.hash = "#" + canonicalId;
  } catch (err) {
    console.error(err);
  }
}
```

Reply:
Added warning above.

Reply:
location.hash contains '#'  so it would have to !document.getElementById(location.hash.substr(1)).

otherwise looks good to me.

Reply:
fixed bug above also.... should actually test code :)